mergify_engine/engine/__init__.py

```diff
@@ -42,39 +42,65 @@
     )
 
 
-async def _check_configuration_changes(ctxt: context.Context) -> bool:
-    if ctxt.pull["base"]["repo"]["default_branch"] == ctxt.pull["base"]["ref"]:
-        ref = None
+async def _check_configuration_changes(
+    ctxt: context.Context,
+    current_mergify_config_file: typing.Optional[context.MergifyConfigFile],
+) -> bool:
+    if ctxt.pull["base"]["repo"]["default_branch"] != ctxt.pull["base"]["ref"]:
+        return False
+
+    config_file_to_validate: typing.Optional[context.MergifyConfigFile] = None
+    prefered_filename = (
+        None
+        if current_mergify_config_file is None
+        else current_mergify_config_file["path"]
+    )
+    # NOTE(sileht): Just a shorcut to do two requests instead of three.
+    if ctxt.pull["changed_files"] <= 100:
         for f in await ctxt.files:
             if f["filename"] in context.Repository.MERGIFY_CONFIG_FILENAMES:
-                ref = typing.cast(
-                    github_types.GitHubRefType, f["contents_url"].split("?ref=")[1]
-                )
-
-        if ref is not None:
-            try:
-                await rules.get_mergify_config(ctxt.repository, ref)
-            except rules.InvalidRules as e:
-                # Not configured, post status check with the error message
-                await ctxt.set_summary_check(
-                    check_api.Result(
-                        check_api.Conclusion.FAILURE,
-                        title="The new Mergify configuration is invalid",
-                        summary=str(e),
-                        annotations=e.get_annotations(e.filename),
-                    )
-                )
-            else:
-                await ctxt.set_summary_check(
-                    check_api.Result(
-                        check_api.Conclusion.SUCCESS,
-                        title="The new Mergify configuration is valid",
-                        summary="This pull request must be merged manually because it modifies Mergify configuration",
-                    )
-                )
+                prefered_filename = f["filename"]
+                break
+        else:
+            return False
+
+    async for config_file in ctxt.repository.iter_mergify_config_files(
+        ref=ctxt.pull["head"]["sha"], prefered_filename=prefered_filename
+    ):
```

Comment:
I don't think we should use preferred_filename here.
In this case:
1. You have `.github/mergify.yml` in the repo already
2. The PR adds `.mergify.yml`

This is going to validate `.github/mergify.yml` but the file that is going to be used in the long run is `.mergify.yml`

Reply:
That's why I check the `path` AND the `sha` changed, to be sure I catch the new file.

Reply:
I even wrote a unit test to check that ;)